cli/command/container/run.go

```diff
@@ -190,7 +190,11 @@ func runContainer(dockerCli *command.DockerCli, opts *runOptions, copts *contain
 		}
 
 		close, err := attachContainer(ctx, dockerCli, &errCh, config, createResponse.ID)
-		defer close()
+
+		if close != nil {
+			defer close()
+		}
```

Comment:
@dnephin -  okay, i thought that we might have some error and still have non nil close.

Thanks for the hint.